### ISSUE TITLE
Time difference while storing avro messages to hdfs directory

### ISSUE THREAD

**Issue by sridharxsridga** [confluentinc/kafka-connect-hdfs#356]
Hi ,

I am trying to find the latency of the hdfs sink connector. I have overall 14 00 000 messages
I triggered the curl command to create hdfs sink connector with flush size as 2 00 000.
During the first time it took almost 11 minutes to complete the task.
After checking logs i got to know that i had other connectors running , So in the next round of testing I deleted all other connectors and deleted the hdfs directory, log directory, +tmp directory.
I deleted the previous connector as well and ran the curl command with new connector name and flush size set to same 2 00 000 messages. This time within 20 seconds all the messages were in the hdfs directory. Could someone please explain why there is such strange time gap for processing . first time it took around 11 minutes for the same topic and in the next test it processed within 20 seconds

**Comment by cricket007** [comment]
@sridharxsridga

First, I'm not sure Avro is important here. 

Secondly, sounds like you're asking for monitoring consumer lag, which can be accomplished by JMX, Confluent Control Center, or Burrow. 

> all the messages were in the hdfs directory

Are you sure all the messages from the beginning were stored? For example, if you didn't reset the consumer group then re-running connector would only grab messages sent from the last time you ran it.

Basically, I would suggest trying to make an isolated test case, but yes, running other tasks in the cluster can affect throughput.
